src/mapping/query/function/split.rs

```diff
@@ -0,0 +1,247 @@
+use super::prelude::*;
+use bytes::Bytes;
+use lazy_static::lazy_static;
+use std::collections::BTreeMap;
+use std::ops::Deref;
+
+#[derive(Debug)]
+pub(in crate::mapping) struct SplitFn {
+    path: Box<dyn Function>,
+    pattern: Box<dyn Function>,
+    limit: Option<Box<dyn Function>>,
+}
+
+impl SplitFn {
+    #[cfg(test)]
+    pub(in crate::mapping) fn new(
+        path: Box<dyn Function>,
+        pattern: Box<dyn Function>,
+        limit: Option<Box<dyn Function>>,
+    ) -> Self {
+        Self {
+            path,
+            pattern,
+            limit,
+        }
+    }
+}
+
+lazy_static! {
+    static ref I_FLAG: Value = Value::from(Bytes::from("i"));
+    static ref G_FLAG: Value = Value::from(Bytes::from("g"));
+    static ref M_FLAG: Value = Value::from(Bytes::from("m"));
+}
+
+/// Create a regex from a map containing fields :
+/// pattern - The regex pattern
+/// flags   - flags including i => Case insensitive, g => Global, m => Multiline.
+fn regex_from_map(mut map: BTreeMap<String, Value>) -> Result<(regex::Regex, bool)> {
```

Comment:
I feel like we can move this into a newtype, and have it impl `From` etc. for a more conventional internal API.